src/http/headers.cr

```diff
@@ -118,64 +179,192 @@ struct HTTP::Headers
     false
   end
 
-  def add(key, value : String)
-    check_invalid_header_content value
+  # Appends a value to the key of a header and returns `self`.  If there is an existing value for the key the value will be converted to an `Array`
+  # `ArgumentError` will be raised if header includes invalid characters.
+  #
+  # ```
+  # require "http/headers"
+  #
+  # headers = HTTP::Headers{"accept-encoding" => "text/html"}
+  # headers.add("accept-encoding", "application/json") # => HTTP::Headers{"accept-encoding" => ["text/html", "application/json"]}
+  # ```
+  def add(key, value : String) : self
+    check_invalid_header_content(value)
     unsafe_add(key, value)
     self
   end
 
-  def add(key, value : Array(String))
+  # Inserts a key value pair into the header collection and returns `self`.
```

Comment:
```suggestion
  # Inserts a key-value pair into the header collection and returns `self`.
```

This description IMO doesn't add any value, you can imply that from the method declaration already.